time stamp
int32 existence_count
int32 distance
char name
